CMakeLists.txt

```diff
@@ -107,3 +107,30 @@ install(
     DESTINATION
     ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
+
+option(DOCUMENTATION "Build the documentation" OFF)
+IF(DOCUMENTATION)
+  find_program(HAS_HOTDOC hotdoc)
```

Comment:
Checking the [docs for find_program()](https://cmake.org/cmake/help/latest/command/find_program.html) I understand that the `HAS_HOTDOC` variable will be set to the full path of the program, when found. I think this would be nicer as follows:

```cmake
    find_program(HOTDOC hotdoc)
    if (NOT HOTDOC)
        message(FATAL_ERROR "hotdoc not found!")
    endif()

    execute_process(
        COMMAND ${HOTDOC} --has-extension c-extension
        RESULT_VARIABLE HAS_HOTDOC_C_EXTENSION
    )
    # ...
```

Note how the `${HOTDOC}` variable is used, this way the path detected by CMake is being used. This can be important (e.g. when cross-building) because the binary to run may not be the one in one of the locations from the `PATH` environment variable.

Reply:
Done.